internal/repos/webhook_build_handler.go

```diff
@@ -2,21 +2,129 @@ package repos
 
 import (
 	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"net/url"
 
 	"github.com/sourcegraph/log"
 
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/globals"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc/auth"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc/github"
+	"github.com/sourcegraph/sourcegraph/internal/httpcli"
+	webhookworker "github.com/sourcegraph/sourcegraph/internal/repos/webhookworker"
+	"github.com/sourcegraph/sourcegraph/internal/types"
 	"github.com/sourcegraph/sourcegraph/internal/workerutil"
 	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"github.com/sourcegraph/sourcegraph/schema"
 )
 
 type webhookBuildHandler struct {
 	store Store
+	doer  httpcli.Doer
 }
 
-func newWebhookBuildHandler(store Store) *webhookBuildHandler {
-	return &webhookBuildHandler{store: store}
+func newWebhookBuildHandler(store Store, doer httpcli.Doer) *webhookBuildHandler {
+	return &webhookBuildHandler{store: store, doer: doer}
 }
 
 func (w *webhookBuildHandler) Handle(ctx context.Context, logger log.Logger, record workerutil.Record) error {
-	return errors.New("TODO")
+	job, ok := record.(*webhookworker.Job)
+	if !ok {
+		return errors.Newf("expected Job, got %T", record)
```

Comment:
What happens if this returns an error? Does the record get deleted or no?

Reply:
Given that I'm not implementing the `nonRetryable` interface, then my job is not `IsNonRetryable` (notice the double negative), so according to [this doc](https://docs.sourcegraph.com/dev/background-information/workers#hook-3-handle-required), it should simply be deleted. With that said, I'm still looking into the purpose of [this](https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Asourcegraph%2Fsourcegraph.git&branch=webhookBuilder&file=internal%2Frepos%2Fwebhookworker%2Fworker.go&editor=VSCode&version=2.2.7&start_row=47&start_col=2&end_row=47&end_col=23&utm_campaign=vscode-extension&utm_medium=direct_traffic&utm_source=vscode-extension&utm_content=vsce-commands)

Reply:
To be sure, you can do it like this:

```suggestion
		return errcode.MakeNonRetryable(errors.Newf("expected Job, got %T", record))
```

This will also require an import `"github.com/sourcegraph/sourcegraph/internal/errcode"`

Reply:
Do we always want to make it non-retryable? Or do we want to retry a set number of times?

Reply:
in this case yes, because we will never succeed in casting the record to `webhookworker.Job` if it is not of that type